src/main/scala/is/hail/expr/ir/PruneDeadFields.scala

```diff
@@ -77,6 +77,27 @@ object PruneDeadFields {
     }
   }
 
+  def minimizeColValues(mv: MatrixValue, valueIR: IR, isArray: Boolean = false): (Type, BroadcastIndexedSeq, IR) = {
+    val matrixType = mv.typ
+    val originalCols = mv.colValues
+    val memo = Memo.empty[BaseType]
+    val valueIRCopy = valueIR.deepCopy()
+    val colDep = memoizeValueIR(valueIRCopy, valueIR.typ, memo)
+      .m.mapValues(_._2)
+      .getOrElse("sa", if (isArray) TArray(TStruct()) else TStruct())
+    log.info(s"minimized col values:\n  From: ${ matrixType.colType }\n  To: ${ colDep }")
```

Comment:
consider setting `val oldColType = matrixType.colType` above as you use it in three places.
do you really want to log this even when no reduction is done?